src/main/java/duke/helpers/Parser.java

```diff
@@ -0,0 +1,117 @@
+package duke.helpers;
+
+import duke.exceptions.OutOfIndexException;
+import duke.exceptions.VagueInputException;
+import duke.exceptions.WrongBooleanException;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+import java.io.PrintWriter;
+import java.util.Scanner;
+
+/**
+ * Handles input cases by the user.
+ */
+public class Parser {
+
+    /**
+     * Handles inputs, and outputs the according string.
+     *
+     * @param sc Scans the inputs of the user.
+     * @param tasks Stores the Tasks objects into
+     * @param pw Writes to the screen for the user to comprehend.
+     */
+    public static void handleInputs(Scanner sc, TaskList tasks, PrintWriter pw) {
+        String input = sc.nextLine();
+        while (!input.equals("bye")) {
+            String[] splitArr = input.split(" ");
+            try {
+                if (input.equals("list")) {
+                    System.out.println(Ui.formatStr(tasks.listThings()));
+                } else if (splitArr.length == 1) {
+                    throw new VagueInputException("Oh no! What do you mean? \n" +
+                            "I'm confused. Please specify... @.@");
+                }
+                else if (splitArr[0].equals("mark") || splitArr[0].equals("unmark")) {
+                    if ((Integer.parseInt(splitArr[1])) > tasks.getSize()) {
+                        throw new OutOfIndexException("Help! \n" +
+                                "The number has to be within range of our task-list!\n" +
+                                "try again.");
+                    }
+                    if (splitArr[0].equals("mark") &&
```

Comment:
I like that you handle the case where the user tries to mark something which is already done again